src/components/form/adapters/base-adapter.ts

```diff
@@ -0,0 +1,211 @@
+import React from 'react';
+import { isNil, isNumber, isString, isBoolean } from 'lodash-es';
+
+const toDashCase = (myStr) =>
+    myStr.replace(/([A-Z])/g, (g) => `-${g[0].toLowerCase()}`);
+
+// Checks if value is a primitive
+const isPrimitive = (value) =>
+    isNil(value) || isNumber(value) || isString(value) || isBoolean(value);
+
+/**
+ * Renders a webcomponent by name
+ */
+export class LimeElementsAdapter extends React.Component<any, any> {
+    component = null;
+
+    static defaultProps = {
+        // The events to bind the the web component
+        events: {},
+        // The props to pass to the web component
+        elementProps: {},
+    };
+
+    constructor(
+        public props: {
+            name: string;
+            events?: any;
+            elementProps?: any;
+        }
+    ) {
+        super(props);
+
+        this.state = {
+            prevEvents: {},
+        };
+
+        this.component = React.createRef();
+
+        this.setComponentProperty = this.setComponentProperty.bind(this);
+    }
+
+    // Sets a property on the web component
+    setComponentProperty([key, value]: [string, any]) {
```

Comment:
I get that this is invoked in the loop further down when getting the entries from an object, but it looks a bit weird to have the signature like this. Do you think we can improve it? Something like
```typescript
setComponentProperty(key: string, value: any)
```
Maybe it will be natural because of some other proposed changes? :thinking: